src/Arcus.Scripting.Management/Scripts/Remove-AzApiManagementSoftDeletedService.ps1

```diff
@@ -0,0 +1,38 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the API Management instance is required"),
+    [Parameter(Mandatory = $false)][string] $SubscriptionId = "",
+    [Parameter(Mandatory = $false)][string] $AccessToken = ""
+)
+
+if($SubscriptionId -eq "" -or $AccessToken -eq ""){
+    # Request accessToken in case the script contains records
+    $token = Get-AzCachedAccessToken
+
+    $AccessToken = $token.AccessToken
+    $SubscriptionId = $token.SubscriptionId
+}
+
+$authHeader = @{
+   'Authorization'='Bearer ' + $AccessToken
+}
+
+Write-Host "Checking if the API Management instance with name '$Name' is listed as a soft deleted service"
+$getUri = 'https://management.azure.com/subscriptions/{0}/providers/Microsoft.ApiManagement/deletedservices?api-version=2021-08-01' -f $SubscriptionId
```

Comment:
We can make this API version also an optional parameter. Like we did with the Logic Apps implementation, too.

Reply:
Will set it up as an optional parameter and set it as default on `2021-08-01`.
I do wonder how we will handle newer api-versions that might have breaking changes, different to Logic Apps we are using a body in one of the http calls so this might be a bit more subject to breaking changes. @stijnmoreels what is your view on this?

Reply:
Ah, you have a point there. Hmm, 🤔 . Since it's already a 'fairly' recent API version, we may choose to not have an optional parameter, then. And only make the Azure cloud configurable. Hmm, that's somewhat the trade-off I guess here, bc there's no built-in PS commands for us to do this.
Yeah, if we don't expose the API version, then we may be able to have more control to update the body/URL when breaking changes occur?
Adding @fgheysels @mbraekman to this.

Reply:
We can make it optional and put in some documentation that states we have tested support up till api-version `2021-08-01`.
Adding this optional parameter does have the benefit of not having to continually update the script whenever new api-versions are available.

Reply:
Aha, yeah, that seems like a good compromise between the two things. Let's do that! 👍

Reply:
It will be better that way bc it will be more streamlined with our other scripts.

Reply:
Added as an optional parameter 👍🏻 see latest commit